frontend/src/pages/modelCatalog/screens/RegisterCatalogModel.tsx

```diff
@@ -207,8 +207,8 @@ const RegisterCatalogModel: React.FC = () => {
           </BreadcrumbItem>
         </Breadcrumb>
       }
-      loaded={modelCatalogSources.loaded}
-      loadError={modelCatalogSources.error}
+      loaded={modelCatalogSources.loaded && registeredModelsLoaded}
+      loadError={modelCatalogSources.error || registeredModelsLoadError}
```

Comment:
This makes the entire page go back to a spinner whenever you change the selected model registry. I wonder if we can instead just show a skeleton in the form fields that require this data (just the name field I think)?